src/main/java/pl/touk/parking/payment/AbstractPaymentPolicy.java

```diff
@@ -0,0 +1,30 @@
+package pl.touk.parking.payment;
+
+import pl.touk.parking.PaymentPolicy;
+import pl.touk.parking.domain.Money;
+
+import java.math.BigDecimal;
+
+public abstract class AbstractPaymentPolicy implements PaymentPolicy {
+
+    @Override
+    public Money calculatePayment(int hours) {
+        if (hours == 1)
+            return firstHourPayment();
+        if (hours == 2)
+            return secondHourPayment();
+        return secondHourPayment().multiply(multiplier(hours));
+    }
+
+    private Money secondHourPayment() {
+        return Money.of(BigDecimal.valueOf(2));
```

Comment:
Czyli klasy dziedziczące tej metody już nie mogą nadpisać? Trochę blokuje to dodawanie nowych cenników

Reply:
Metoda jest prywatna, bo na tę chwilę nie potrzebuje większej widoczności. Jeśli zajdzie taka potrzeba można to łatwo zmienić. Gdyby to była biblioteka inaczej był do tego podszedł.